src/Spatial/Euclidean/LIneSegment2D.cs

```diff
@@ -0,0 +1,224 @@
+        /// <returns>A line segment from the first point to the second point.</returns>
+        public static LineSegment2D Parse(string startPointString, string endPointString)
+        {
+            return new LineSegment2D(Point2D.Parse(startPointString), Point2D.Parse(endPointString));
+        }
+
+        /// <summary>
+        /// Translates a line according to a provided vector
+        /// </summary>
+        /// <param name="vector">A vector to apply</param>
+        /// <returns>A new translated linesegment</returns>
+        public LineSegment2D Translate(Vector2D vector)
+        {
+            var startVector = this.StartPoint.ToVector2D().Add(vector);
+            var endVector = this.EndPoint.ToVector2D().Add(vector);
+            return new LineSegment2D(new Point2D(startVector.X, startVector.Y), new Point2D(endVector.X, endVector.Y));
+        }
+
+        /// <summary>
+        /// Returns a new linesegment between the clostes point on this line segment and a point.
+        /// </summary>
+        /// <param name="p">the point to create a line to</param>
+        /// <returns>A linesegment between the point and the nearest point on this segment.</returns>
+        [Pure]
+        public LineSegment2D LineTo(Point2D p)
+        {
+            return new LineSegment2D(this.ClosestPointTo(p), p);
+        }
+
+        /// <summary>
+        /// Returns the closest point on the line to the given point.
+        /// </summary>
+        /// <param name="p">The point that the returned point is the closest point on the line to</param>
+        /// <returns>The closest point on the line to the provided point</returns>
+        [Pure]
+        public Point2D ClosestPointTo(Point2D p)
+        {
+            var v = this.StartPoint.VectorTo(p);
+            var dotProduct = v.DotProduct(this.Direction);
+
+            if (dotProduct < 0)
+            {
+                dotProduct = 0;
+            }
+
+            var l = this.Length;
+            if (dotProduct > l)
+            {
+                dotProduct = l;
+            }
+
+            var alongVector = dotProduct * this.Direction;
+            return this.StartPoint + alongVector;
+        }
+
+        /// <summary>
+        /// Compute the intersection between two lines if the angle between them is greater than a specified
+        /// angle tolerance.
+        /// </summary>
+        /// <param name="other">The other line to compute the intersection with</param>
+        /// <param name="intersection">When this method returns, contains the intersection point, if the conversion succeeded, or the default point if the conversion failed.</param>
+        /// <param name="tolerance">The tolerance used when checking if the lines are parallel</param>
+        /// <returns>True if an intersection exists; otherwise false</returns>
+        [Pure]
+        public bool TryIntersect(LineSegment2D other, out Point2D intersection, Angle tolerance)
```

Comment:
We can probably add an Intersect method that throws on failure that calls this internally later. But that is another PR and we need to bikeshed it some due to existing Intersect methods returning `Nullable<Point2D>`